src/runtime_view_proxy.js

```diff
@@ -0,0 +1,187 @@
+      return createTypedArray(BigUint64Array, offset, length);
+    default:
+      throw new Error('Invalid type');
+  }
+}
+
+function createProxyHandler (type, heapBlocks) {
+  const firstHeapBlock = heapBlocks[0]
+  const bpe = firstHeapBlock.BYTES_PER_ELEMENT
+
+  function getRealStartAndEnd(start, end) {
+    const startReal = (start || 0) * bpe
+    const endReal = end ? end * bpe : wasmMemory.byteLength
+    return [startReal, endReal]
+  }
+
+  function copyWithin (target, start, end) {
+    if (target * bpe >= maxArraySize || start * bpe >= maxArraySize || (end && end * bpe >= maxArraySize)) {
+      const len = end - start
+      const targetArray = getHeapBlock(type, target * bpe, len)
+      const sourceArray = getHeapBlock(type, start * bpe, len)
+      targetArray.set(sourceArray)
+      return heapBlocks[0]
+    } else {
+      return heapBlocks[0].copyWithin(target, start, end)
+    }
+  }
+
+  function setOverridden(array, offset) {
+    const offsetReal = (offset || 0) * bpe
+    if (offsetReal >= maxArraySize || array.byteLength + offsetReal >= maxArraySize) {
+      const targetArray = getHeapBlock(type, offsetReal, array.length)
+      targetArray.set(array)
+    } else {
+      firstHeapBlock.set(array, offset)
+    }
+  }
+
+  function subarray(start, end) {
+    const [startReal, endReal] = getRealStartAndEnd(start, end)
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      return getHeapBlock(type, startReal, endReal - startReal)
+    } else {
+      return firstHeapBlock.subarray(start, end)
+    }
+  }
+
+  function fill(value, start, end) {
+    const [startReal, endReal] = getRealStartAndEnd(start, end)
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      const hb = getHeapBlock(type, startReal, endReal - startReal)
+      hb.fill(value, 0, end - start)
+      return firstHeapBlock
+    } else {
+      return firstHeapBlock.fill(value, start, end)
+    }
+  }
+  function slice(start, end) {
+    const [startReal, endReal] = getRealStartAndEnd(start, end)
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      const hb = getHeapBlock(type, startReal, endReal - startReal)
+      return hb.slice(start, end)
+    } else {
+      return firstHeapBlock.slice(start, end)
+    }
+  }
+
+  return {
+    get(target, property) {
+      if (parseInt(property, 10) == property) {
+        const memoryOffset = property * bpe
+        const blockNumber = Math.floor(memoryOffset / maxArraySize)
+        return heapBlocks[blockNumber][property - blockNumber * maxArraySize]
+      }
+
+      if (property === 'copyWithin') {
+        return copyWithin
+      }
+
+      if (property === 'set') {
+        return setOverridden
+      }
+
+      if (property === 'subarray') {
+        return subarray
+      }
+
+      if (property === 'fill') {
+        return fill
+      }
+
+      if (property === 'slice') {
+        return slice
+      }
+
+      return firstHeapBlock[property]
```

Comment:
It would it be better to fail here?  i.e. is it risky to fall back to getting props from the first block only?   What kind of properties get this far?

Reply:
at least .buffer